scalafmt-core/shared/src/main/scala/org/scalafmt/util/TokenOps.scala

```diff
@@ -53,16 +53,21 @@ object TokenOps {
   def shouldGet2xNewlines(
       tok: FormatToken
   )(implicit style: ScalafmtConfig): Boolean =
-    !forceDocstringBlankLine(tok.left, style) && {
-      tok.hasBlankLine ||
-      (forceDocstringBlankLine(tok.right, style) && !tok.left.is[Comment])
-    }
+    tok.hasBlankLine || blankLineBeforeDocstring(tok.left, tok.right)
+
+  def blankLineBeforeDocstring(
+      left: Token,
+      right: Token
+  )(implicit style: ScalafmtConfig): Boolean =
+    right.is[Token.Comment] &&
+      blankLineBeforeDocstring(left, right.asInstanceOf[Token.Comment])
```

Comment:
What about

```scala
  def blankLineBeforeDocstring(
      left: Token,
      right: Token
  )(implicit style: ScalafmtConfig): Boolean =
    right match {
      case comment: Token.Comment =>
        style.optIn.forceNewlineBeforeDocstringSummary &&
          !left.is[Token.Comment] &&
          comment.syntax.startsWith("/**")
      case _ => false
    }
```
?
one function instead of two, get rid of `asInstanceOf`

Reply:
i did it on purpose, i will use it in another commit to change the `Router` match for comments.